test/common/access_log/access_log_impl_test.cc

```diff
@@ -918,6 +921,150 @@ name: envoy.file_access_log
+      "\"ABORTED\" \"OUT_OF_RANGE\" \"UNIMPLEMENTED\" \"INTERNAL\" \"UNAVAILABLE\" \"DATA_LOSS\" "
+      "\"UNAUTHENTICATED\"]]): grpc_status_filter {\n  statuses: \"NOT_A_VALID_CODE\"\n}\n");
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterPass) {
+  const std::string yaml = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+      - OK
+config:
+  path: /dev/null
+  )EOF";
+
+  const InstanceSharedPtr log =
+      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_);
+
+  response_trailers_.addCopy(Http::Headers::get().GrpcStatus, "0");
+
+  EXPECT_CALL(*file_, write(_));
+  log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterBlock) {
+  const std::string yaml = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+      - OK
+config:
+  path: /dev/null
+  )EOF";
+
+  const InstanceSharedPtr log =
+      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_);
+
+  response_trailers_.addCopy(Http::Headers::get().GrpcStatus, "1");
+
+  EXPECT_CALL(*file_, write(_)).Times(0);
+  log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterHttpCodes) {
+  const std::string yamlPrefix = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+)EOF";
+
+  const std::string yamlSuffix = R"EOF(
+config:
+  path: /dev/null
+  )EOF";
+
+  // This mapping includes UNKNOWN <-> 200 because we expect that gRPC should provide an explicit
+  // status code for successes. In general, the only status codes that receive an HTTP mapping are
+  // those enumerated below with a non-UNKNOWN mapping. See: //source/common/grpc/status.cc and
+  // https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md.
+  const std::vector<std::tuple<std::string, uint64_t>> statusMapping({{"UNKNOWN", 200},
+                                                                      {"INTERNAL", 400},
+                                                                      {"UNAUTHENTICATED", 401},
+                                                                      {"PERMISSION_DENIED", 403},
+                                                                      {"UNIMPLEMENTED", 404},
+                                                                      {"UNAVAILABLE", 504}});
+
+  std::ostringstream out;
+  for (const auto& pair : statusMapping) {
+    out << yamlPrefix << "      - " << std::get<0>(pair) << yamlSuffix;
+    stream_info_.response_code_ = std::get<1>(pair);
+
+    const InstanceSharedPtr log =
+        AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(out.str()), context_);
+
+    EXPECT_CALL(*file_, write(_));
+    log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
+
+    out.clear();
+  }
+}
+
```

Comment:
can we also add a test that then defaults to unknown if no grpc code nor http status code is present?

Reply:
Will do.